lib/engine/route.rb

```diff
@@ -38,13 +38,15 @@ def connections
     end
 
     def next_connection(node, connection, other)
-      connections = select(node, other)
+      connections = select(node, other, connection)
       index = connections.find_index(connection) || connections.size
       connections[index + 1]
     end
 
-    def select(node, other)
+    def select(node, other, keep = nil)
       other_paths = @routes.reject { |r| r == self }.flat_map(&:paths)
+      other_paths.concat(@connections.reject { |c| c[:connection] == keep }
+                          .flat_map { |c| c[:connection].paths })
```

Comment:
i don't understand this code

c[:connection] is a connection object, how can that every equal an array of paths?

also line 52 !@connection.include?(c) looks like it's a bug, can you fix that?

Reply:
I'm filtering out the connection that matches keep first, then calling flat_map on all of the paths in the rest of the connections in @connections.

Reply:
Pretty sure if line 52 worked as intended, it would break being able to cycle paths. The intent, though is to do exactly what my PR is trying to do.  I need to think about whether is is a better way.

Reply:
Scratch that. I don't know exactly what the intent was, but if I fix it to actually work it breaks cycling paths. It's preventing from selecting a connection already in the connection list, which is OK, except when you'd like to cycle paths via "next_connection". My PR will do what the intent was (except cycling works). So I'm just tossing the old code.